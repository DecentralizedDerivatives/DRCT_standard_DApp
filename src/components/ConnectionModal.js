import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
import { Link } from 'react-router-dom';
import { showConnectionModal } from '../actions/statusActions';

// Refactor as class component
class ConnectionModal extends Component {
  constructor(props) {
    super(props);

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.props.showConnectionModal(false);
    // this.props.history.push('/');
  }

  render() {
    return (
      <Modal isOpen={this.props.isConnectModalOpen}>
        <ModalHeader id="alert-dialog-title">
          Unable to detect network
        </ModalHeader>
        <ModalBody>
          <div id="alert-dialog-description">
            <span>
              Are you sure you're connected to the Ethereum Main Network?
            </span>
          </div>
        </ModalBody>
        <ModalFooter>
          <button className="btn btn-primary" onClick={this.handleClick}>
            <Link className="button-link" to="/">
              Homepage
            </Link>
          </button>
        </ModalFooter>
      </Modal>
    );
  }
}

ConnectionModal.propTypes = {
  showConnectionModal: PropTypes.func.isRequired,
  isConnectModalOpen: PropTypes.bool.isRequired
};

const mapStateToProps = state => ({
  isConnectModalOpen: state.status.isConnectModalOpen
});

export default connect(
  mapStateToProps,
  { showConnectionModal }
)(ConnectionModal);
