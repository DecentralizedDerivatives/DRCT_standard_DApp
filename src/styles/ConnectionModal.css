.button-link {
  text-decoration: none;
  color: #3f51b5;
}

.button-text {
  color: #fff;
}

.button {
  width: 100%;
  background-color: #114770;
}

.button:hover {
  background-color: #9fb5c5;
}

.button-disabled: {
  width: 100%;
  background-color: #9fb5c5;
}
