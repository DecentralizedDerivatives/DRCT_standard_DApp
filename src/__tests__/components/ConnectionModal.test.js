import ConnectionModal from '../../components/ConnectionModal';

describe('<ConnectionModal />', () => {
  it('renders the component', () => {
    const wrapper = shallow(<ConnectionModal store={initStore()} />).dive();

    expect(wrapper).toMatchSnapshot();

    wrapper.find('ModalFooter > button').simulate('click');
  });
});
