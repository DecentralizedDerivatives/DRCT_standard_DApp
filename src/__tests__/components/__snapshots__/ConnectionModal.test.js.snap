// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConnectionModal /> renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConnectionModal
    isConnectModalOpen={false}
    showConnectionModal={[Function]}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoFocus": true,
      "backdrop": true,
      "backdropTransition": Object {
        "mountOnEnter": true,
        "timeout": 150,
      },
      "centered": false,
      "children": Array [
        <ModalHeader
          charCode={215}
          closeAriaLabel="Close"
          id="alert-dialog-title"
          tag="h5"
          wrapTag="div"
        >
          Unable to detect network
        </ModalHeader>,
        <ModalBody
          tag="div"
        >
          <div
            id="alert-dialog-description"
          >
            <span>
              Are you sure you're connected to the Ethereum Main Network?
            </span>
          </div>
        </ModalBody>,
        <ModalFooter
          tag="div"
        >
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            <Link
              className="button-link"
              replace={false}
              to="/"
            >
              Homepage
            </Link>
          </button>
        </ModalFooter>,
      ],
      "fade": true,
      "isOpen": false,
      "keyboard": true,
      "modalTransition": Object {
        "timeout": 300,
      },
      "onClosed": [Function],
      "onOpened": [Function],
      "role": "dialog",
      "zIndex": 1050,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "charCode": 215,
          "children": "Unable to detect network",
          "closeAriaLabel": "Close",
          "id": "alert-dialog-title",
          "tag": "h5",
          "wrapTag": "div",
        },
        "ref": null,
        "rendered": "Unable to detect network",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div
            id="alert-dialog-description"
          >
            <span>
              Are you sure you're connected to the Ethereum Main Network?
            </span>
          </div>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              Are you sure you're connected to the Ethereum Main Network?
            </span>,
            "id": "alert-dialog-description",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Are you sure you're connected to the Ethereum Main Network?",
            },
            "ref": null,
            "rendered": "Are you sure you're connected to the Ethereum Main Network?",
            "type": "span",
          },
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            <Link
              className="button-link"
              replace={false}
              to="/"
            >
              Homepage
            </Link>
          </button>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="button-link"
              replace={false}
              to="/"
            >
              Homepage
            </Link>,
            "className": "btn btn-primary",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Homepage",
              "className": "button-link",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Homepage",
            "type": [Function],
          },
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoFocus": true,
        "backdrop": true,
        "backdropTransition": Object {
          "mountOnEnter": true,
          "timeout": 150,
        },
        "centered": false,
        "children": Array [
          <ModalHeader
            charCode={215}
            closeAriaLabel="Close"
            id="alert-dialog-title"
            tag="h5"
            wrapTag="div"
          >
            Unable to detect network
          </ModalHeader>,
          <ModalBody
            tag="div"
          >
            <div
              id="alert-dialog-description"
            >
              <span>
                Are you sure you're connected to the Ethereum Main Network?
              </span>
            </div>
          </ModalBody>,
          <ModalFooter
            tag="div"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
            >
              <Link
                className="button-link"
                replace={false}
                to="/"
              >
                Homepage
              </Link>
            </button>
          </ModalFooter>,
        ],
        "fade": true,
        "isOpen": false,
        "keyboard": true,
        "modalTransition": Object {
          "timeout": 300,
        },
        "onClosed": [Function],
        "onOpened": [Function],
        "role": "dialog",
        "zIndex": 1050,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "charCode": 215,
            "children": "Unable to detect network",
            "closeAriaLabel": "Close",
            "id": "alert-dialog-title",
            "tag": "h5",
            "wrapTag": "div",
          },
          "ref": null,
          "rendered": "Unable to detect network",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <div
              id="alert-dialog-description"
            >
              <span>
                Are you sure you're connected to the Ethereum Main Network?
              </span>
            </div>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                Are you sure you're connected to the Ethereum Main Network?
              </span>,
              "id": "alert-dialog-description",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Are you sure you're connected to the Ethereum Main Network?",
              },
              "ref": null,
              "rendered": "Are you sure you're connected to the Ethereum Main Network?",
              "type": "span",
            },
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <button
              className="btn btn-primary"
              onClick={[Function]}
            >
              <Link
                className="button-link"
                replace={false}
                to="/"
              >
                Homepage
              </Link>
            </button>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="button-link"
                replace={false}
                to="/"
              >
                Homepage
              </Link>,
              "className": "btn btn-primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Homepage",
                "className": "button-link",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Homepage",
              "type": [Function],
            },
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
